[v1.5][F12-B3] Media Socializer

shadow2496 opened the pull request:
@Nethergale @KevinChuangCH

OuyangDanwen commented on README.adoc:
```diff
@@ -1,10 +1,9 @@
-https://ci.appveyor.com/project/damithc/addressbook-level4[image:https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true[Build status]]
-https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master[Coverage Status]]
-https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&utm_medium=referral&utm_content=se-edu/addressbook-level4&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a[Codacy Badge]]
+https://travis-ci.org/CS2103JAN2018-F12-B3/main[image:https://travis-ci.org/CS2103JAN2018-F12-B3/main.svg?branch=master[Build Status]]
```
Good.

OuyangDanwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -864,6 +959,9 @@ Windows, Linux, Unix, OS-X
 A contact detail that is not meant to be shared with others
 
+[[social-media-platform]] Social media platform::
+A platform that makes people distribute information and encourages them to connect with others
```
You have to update the product survey.

OuyangDanwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,6 +903,38 @@ Use case ends.
+
+*MSS*
+
+1.  User requests to apply the filter of <<social-media-platform,a social media platform>>
```
Write it in a A-B-A-B format instead of A-A-B. This is required. It should be a forth-back interaction.

OuyangDanwen reviewed (commented):
Please update the UG asap. You should document all features (up to V2.0) in UG in detail. Mark them accordinly, eg. V1.5, V2.0, etc. This is required for the team component last week. Therefore, I have to consider your team component a miss for last week. @shadow2496 @KevinChuangCH @Nethergale  In addition, update the AboutUs Page and ContactUS page asap by next tutorial.

KevinChuangCH commented:
@OuyangDanwen 
Sorry, the update of AboutUs and part of update of UG was actually done. I forgot to merge them since I thought the tutor has check for them during the tutorial.

OuyangDanwen reviewed (commented):
UG is generally okay. Make sure it is ready before peer testing. DG needs more work. Please update the diagrams accordingly to meet your design. It is important that you have diagrams to explain your design. It is also good to show them in your individual portfolio. Do not only use plain words. Apply your UML knowledge as far as possible. @shadow2496 @KevinChuangCH @Nethergale

Nethergale committed:
Update custom person creation parameters to hard-coded values

Nethergale committed:
Update xml files for XmlUtilTest
Fix checkstyle errors

Nethergale committed:
Update user guide and app title in config file

Nethergale committed:
Re-add config.json to gitignore

Nethergale committed:
Fix formatting in user guide
Add new line to config file

Nethergale committed:
Fix listing format in user guide
Add config.json back to gitignore

Nethergale committed:
Test fixing format in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test fixing auto links in user guide

Nethergale committed:
Test using table

Nethergale committed:
Test formatting using table

Nethergale committed:
Test formatting using table

Nethergale committed:
Test formatting using table

Nethergale committed:
Test putting table into note

Nethergale committed:
Format table in note

Nethergale committed:
Change formatting of table in note

Nethergale committed:
Change formatting of table in note

Nethergale committed:
Change attributes of table

Nethergale committed:
Change headers of table

Nethergale committed:
Add test cases for add platform command parser

Nethergale committed:
Add new line to config file

Nethergale committed:
Remove config file from gitignore

Nethergale committed:
Re-add config file to gitignore

Nethergale committed:
Untrack config file

Nethergale committed:
Update application version and application name

Nethergale committed:
Update portfolio

shadow2496 committed:
Add login() in Account

Nethergale committed:
Add authorship for collate

Nethergale committed:
Merge pull request #87 from Nethergale/feature/manageSocialMediaPlatform

Add test cases for add platform command

KevinChuangCH committed:
BrowserPanel.fxml: Edit browser panel from a single Webview to 2 tabs, each with a different Webview.

KevinChuangCH committed:
BrowserPanel.java: Implementation of the browser in the 2nd tab. Dummy search page is used for now.

KevinChuangCH committed:
BrowserPanel.java: Change the name of the constant to improve readability.

KevinChuangCH committed:
BrowserPanelTest.java: Update the change of the constant name.

KevinChuangCH committed:
AddressBookSystemTest.java: Update the change of the constant name.

KevinChuangCH committed:
BrowserPanel.java: Correct coding standard.

KevinChuangCH committed:
BrowserPanelTest.java: Correct coding standard.

KevinChuangCH committed:
AddressBookSystemTest.java: Correct coding standard.

KevinChuangCH committed:
Merge pull request #90 from KevinChuangCH/UI

shadow2496 committed:
Add loginAccount() in AddressBook

Nethergale committed:
Add implementation of remove platform command

Nethergale committed:
Fix checkstyle errors

shadow2496 committed:
Update LoginCommand.execute() to generate the login dialog url

Nethergale committed:
Add test case to address book parser for remove platform command

Nethergale committed:
Add test cases for remove platform command parser
Make minor changes to other tests and user guide

KevinChuangCH committed:
SearchAllCommand.java: Create SearchAllCommand Class, no implementation yet.

KevinChuangCH committed:
SearchAllCommand.java: Add implementation of SearchAllCommand, now the SearchAllCommand should perform the same task as SelectCommand.

KevinChuangCH committed:
SearchAllCommandParser.java: Create new SearchAllCommandParser class, no implementation yet.

KevinChuangCH committed:
SearchAllCommandParser.java: Add implementation of SearchAllCommandParser. The SearchAllCommandParser now performs the same task as SelectCommand, except that it returns a SearchAllCommand object instead of a SelectCommand Object.

KevinChuangCH committed:
AddressBookParser.java: Add the case for SearchAllCommand in the parseCommand method.

Nethergale committed:
Add test cases for remove platform command

Nethergale committed:
Fix checkstyle errors

Nethergale committed:
Merge pull request #91 from Nethergale/feature/manageSocialMediaPlatform

Add implementation of remove platform command

Nethergale committed:
Update SampleDataUtil persons, BrowserPanel urls and BrowserPanelTest

Nethergale committed:
Add collated code files

KevinChuangCH committed:
SearchPersonOnAllPlatformEvent.java: Create new SearchPersonOnAllPlatformEvent class for SearchAllCommand.

KevinChuangCH committed:
SearchAllCommand.java: Implement the actual Implementation of the SearchAllCommand.

KevinChuangCH committed:
SearchAllCommandParser.java: Implement the actual implementation of SearchAllCommandParser.

KevinChuangCH committed:
ParserUtil.java: Add parseSearchName method to check for the validation of the input name.

Nethergale committed:
Merge pull request #96 from Nethergale/master

Add collated code files

KevinChuangCH committed:
BrowserPanel.java: Add new handleSearchPersonOnAllPlatformEvent to handle the search all event. Changed names of some constants to improve readability.

Nethergale committed:
Change image of ui for readme

KevinChuangCH committed:
Merge branch 'master' into SearchOnSocialMedia

Nethergale committed:
Merge pull request #98 from Nethergale/master

Change image of UI for readme

KevinChuangCH committed:
SearchPersonOnAllPlatformEvent.java: Correct coding standard error.

KevinChuangCH committed:
SearchAllCommandParser.java: Correct coding standard error, remove unused import.

KevinChuangCH committed:
Merge branch 'SearchOnSocialMedia'

KevinChuangCH committed:
SearchAllCommandParser.java: Correct coding standard error, remove unused import.

KevinChuangCH committed:
SearchPersonOnAllPlatformEvent.java: Correct coding standard error, add missing white space.

KevinChuangCH committed:
Merge pull request #97 from KevinChuangCH/SearchOnSocialMedia

Search on social media

KevinChuangCH committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B3/main

shadow2496 committed:
Add ShowLoginDialogRequestEvent

KevinChuangCH committed:
SearchFacebookCommand.javac: Create SearchFacebookCommand class, no implementation yet.

KevinChuangCH committed:
SearchFacebookCommand.java: Add implementation of SearchFacebookCommand, now the SearchFacebookCommand should perform the same task as SearchAllCommand.

KevinChuangCH committed:
SearchFacebookCommandParser.java: Create new SearchFacebookCommandParser class, no implementation yet.

KevinChuangCH committed:
SearchFacebookCommandParser.java: Add implementation of SearchFacebookCommandParser. The SearchFacebookCommandParser now performs the same task as SearchAllCommandParser, except that it returns a SearchFacebookCommand object instead of a SearchAllCommand object.

KevinChuangCH committed:
AddressBookParser.java: Add the case for SearchFacebookCommand in the parseCommand method.

KevinChuangCH committed:
SearchPersonOnAllPlatformEvent.java: Create new SearchPersonOnAllPlatformEvent class for SearchAllCommand.

Nethergale committed:
Update user guide and collate code files of group

KevinChuangCH committed:
SearchFacebookCommand.java: Implement the actual Implementation of the SearchFacebookCommand.

KevinChuangCH committed:
BrowserPanel.java: Add new handleSearchPersonOnFacebookEvent to handle the search facebook event.

KevinChuangCH committed:
SearchFacebookCommand.java: Correct MESSAGE_SEARCH_PERSON_SUCCESS, add missing whitespace.

KevinChuangCH committed:
SearchFacebookCommandParser.java: Add author.

KevinChuangCH committed:
SearchPersonOnFacebookEvent.java: Add author.

KevinChuangCH committed:
SearchFacebookCommand.java: Add author.

KevinChuangCH committed:
SearchTwitterCommand.java: Create SearchTwitterCommand class, no implementation yet.

KevinChuangCH committed:
SearchTwitterCommandParser.java: Create new SearchTwitterCommandParser class.

shadow2496 committed:
Add passVerificationCode() in BrowserPanel

KevinChuangCH committed:
SearchPersonOnTwitterEvent.java: Create new SearchPersonOnTwitterEvent class for SearchTwitterCommand.

KevinChuangCH committed:
SearchTwitterCommand.java: Implement the SearchTwitterCommand class.

KevinChuangCH committed:
AddressBookParser.java: Add the case for SearchTwitterCommand in the parseCommand method.

KevinChuangCH committed:
BrowserPanel.java: Add new handleSearchPersonOnTwitterEvent to handle the search twitter event.

KevinChuangCH committed:
Merge pull request #100 from KevinChuangCH/SearchOnFacebook

Search on facebook

KevinChuangCH committed:
Merge branch 'master' into SearchOnTwitter

KevinChuangCH committed:
Update AddressBookParser.java

KevinChuangCH committed:
Update BrowserPanel.java

KevinChuangCH committed:
Merge pull request #101 from KevinChuangCH/SearchOnTwitter

Search on twitter

shadow2496 committed:
Update Logic, Model to pass the verification code to Account

KevinChuangCH committed:
Merge pull request #99 from Nethergale/master

Update user guide and collate code files of group

shadow2496 committed:
Merge branch 'master' into feature/model/login_facebook

shadow2496 committed:
Update the version to v1.4

shadow2496 committed:
Merge pull request #103 from shadow2496/setup/v1.4

Set up v1.4 Media Socializer

KevinChuangCH committed:
UserGuide.adoc: Update the User Guide to inlcude the 3 newly implemented features (searchall, searchfb, and searchtwitter).

KevinChuangCH committed:
UserGuide.adoc: Update User Guide to include information about the use of dummy search page.

shadow2496 committed:
UserGuide: update the login command feature

shadow2496 committed:
Merge pull request #104 from KevinChuangCH/UserGuideForPeerTesting

UserGuide.adoc

shadow2496 committed:
Remove MockUi.png

shadow2496 committed:
Merge pull request #105 from shadow2496/feature/docs/user

UserGuide: updated the login command feature

shadow2496 committed:
Merge pull request #106 from shadow2496/feature/docs/readme

README: removed MockUi.png

Nethergale committed:
Update regex for facebook and twitter links
Add test cases for links and minor changes to if loops in link class

Nethergale committed:
Fix checkstyle errors

Nethergale committed:
Merge pull request #114 from Nethergale/master

Update regex for facebook and twitter links

Nethergale committed:
Update portfolio and developer guide for changes to feature contribution

Nethergale committed:
Make minor corrections to developer guide

Nethergale committed:
Make minor corrections to portfolio

KevinChuangCH committed:
SearchPersonEvent.java: Create new SearchPersonEvent class for abstraction of SearchPersonOnAllPlatformEvent, SearchPersonFacebookEvent, and SearchPersonOnTwitterEvent. No Implementation yet.

KevinChuangCH committed:
SearchCommand.java: Create new SearchCommand class to replace SearchAllCommand, SearchFacebookCommand, and SearchTwitterCommand classes. This is done to reduce similar code.

KevinChuangCH committed:
SearchCommandParser.java: Create new SearchCommandParser class to replace SearchAllCommandParser, SearchFacebookCommandParser, and SearchTwitterCommandParser classes. This is done to reduce similar code.

KevinChuangCH committed:
SearchPersonEvent.java: Implement the SearchPersonEvent

KevinChuangCH committed:
BrowserPanel.java: Create new handleSearchPersonEvent method to replace the handleSearchPersonOnAllPlatformEvent, handleSearchPersonOnFacebookEvent, and handleSearchPersonOnTwitterEvent methods. This is done to abstract out similar codes in the three replaced methods.

KevinChuangCH committed:
SocialMediaPlatform.java: Add MESSAGE_PLATFORM_CONSTRAINTS and isValidPlatform for checking of input platform in search command.

KevinChuangCH committed:
Facebook.java: Add PLATFORM_KEYWORD and PLATFORM_ALIAS to be use in a search person event.

KevinChuangCH committed:
Twitter.java: Add PLATFORM_KEYWORD and PLATFORM_ALIAS to be use in a search person event.

KevinChuangCH committed:
ParserUtil.java: Implement new parsePlatformToSearch method to check for validation of input platform in a search command.

KevinChuangCH committed:
AddressBookParser.java: Add the case for SearchCommand in the parseCommand method.

KevinChuangCH committed:
SocialMediaPlatform.java: Remove extra blank line.

KevinChuangCH committed:
SearchCommandParser.java: Correct mistake in comment.

KevinChuangCH committed:
Remove classes that are replaced by SearchCommand class, SearchCommandParser class, and SearchPersonEvent class.

KevinChuangCH committed:
BrowserPanel.java: Remove unused methods and imports.

KevinChuangCH committed:
AddressBookParser.java: Remove cases that are replaced by SearchCommand.

KevinChuangCH committed:
SearchCommand.java: Edit COMMAND_WORD to prompt the user that PLATFORM is case sensitive.

KevinChuangCH committed:
BrowserPanel.java: Correct coding style to follow the coding standard.

KevinChuangCH committed:
SearchCommandParser.java: Correct coding style to follow the coding standard.

KevinChuangCH committed:
SocialMediaPlatform.java: Correct coding style to follow the coding standard.

KevinChuangCH committed:
Merge pull request #117 from KevinChuangCH/AbstractSimilarCode

Abstract similar code

KevinChuangCH committed:
BrowserPanel.java: Abstract out common code.

KevinChuangCH committed:
Merge pull request #123 from KevinChuangCH/AbstractSimilarCode

BrowserPanel.java: Abstract out common code.

Nethergale committed:
Merge pull request #116 from Nethergale/master

Update portfolio and developer guide for changes to feature contribution

KevinChuangCH committed:
images: Add pictures used in Developer Guide.

KevinChuangCH committed:
DeveloperGuide.adoc: Minor update of Developer Guide.

KevinChuangCH committed:
chuangchengheng.adoc: Update personal portfolio.

KevinChuangCH committed:
SocialMediaPlatform.java: Add USERNAME_VALIDATION_REGEX and MESSAGE_USERNAME_CONSTRAINTS to facilitate checking of input username in the search feature.

KevinChuangCH committed:
ParserUtil.java: Use USERNAME_VALIDATION_REGEX of SocialMediaPlatform to check for validation of input username in search function.

KevinChuangCH committed:
SocialMediaPlatform.java: Correct coding style to follow the coding standard.

KevinChuangCH committed:
SocialMediaPlatform.java: Correct coding style to follow the coding standard.

KevinChuangCH committed:
Merge pull request #124 from KevinChuangCH/CorrectUsernameCheck

Correct username check

Nethergale committed:
Update BrowserPanel to show profiles of persons
Fix existing test cases due to browser panel changes
Add loading page for future use
Extend wait time for browser testing

Nethergale committed:
Change config files to correct application name

Nethergale committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B3/main into feature/manageSocialMediaPlatform

Fix conflict with BrowserPanel.java

Nethergale committed:
Fix checkstyle errors in browser panel

Nethergale committed:
Remove whitespace line

Nethergale committed:
Add comments to methods for browser panel

Nethergale committed:
Change method names while refactoring usages
Add collate code tags

Nethergale committed:
Make changes to typical persons

Nethergale committed:
Merge pull request #126 from Nethergale/feature/manageSocialMediaPlatform

Display social media platforms of selected person

Nethergale committed:
Make minor correction to user guide

Nethergale committed:
Add variable to browser panel

Nethergale committed:
Remove syntax error

Nethergale committed:
Increase wait for event timeout

Nethergale committed:
Edit typical persons and build.gradle

Nethergale committed:
Merge pull request #127 from Nethergale/feature/manageSocialMediaPlatform

Make minor correction to user guide and test CI checks

Nethergale committed:
Add new line

Nethergale committed:
Rewrite some code for testing of URLs

Nethergale committed:
Reduce wait timeout

Nethergale committed:
Update travis.yml

KevinChuangCH committed:
BrowserPanel.java: Included the feature of opening and closing tab into the search feature. Replace DUMMY_PROFILE_PAGE_URL with defaultPage.

KevinChuangCH committed:
AddressBookSystemTest.java: Replace DUMMY_PROFILE_PAGE_URL with the defaultPage.

Nethergale committed:
Merge pull request #128 from Nethergale/feature/manageSocialMediaPlatform

Test CI checks

KevinChuangCH committed:
Merge branch 'master' into UpdateSearchFeature

KevinChuangCH committed:
Update BrowserPanel.java

KevinChuangCH committed:
.travis.yml: Attempt to fix Travis.

KevinChuangCH committed:
BrowserPanel.java: Fix coding style error.

KevinChuangCH committed:
.travis.yml: Redo previous change.

KevinChuangCH committed:
AddressBookSystemTest.java: Fix coding style error, remove unused import.

KevinChuangCH committed:
Attempt to fix Travis.

KevinChuangCH committed:
BrowserPanel.java: Attempt to fix failing test cases on Travis.

KevinChuangCH committed:
Attempt to fix failing test cases on Travis.

KevinChuangCH committed:
images: Add images used in Developer Guide

KevinChuangCH committed:
UserGuide.adoc: Update User Guide.

KevinChuangCH committed:
DeveloperGuide.adoc: Update Developer Guide.

KevinChuangCH committed:
images: Add new images used in Developer Guide

KevinChuangCH committed:
DeveloperGuide.adoc: Minor update, included screen shot of the application.

KevinChuangCH committed:
images: Add images used in Developer Guide.

KevinChuangCH committed:
DeveloperGuide.adoc: Update Developer Guide.

KevinChuangCH committed:
BrowserPanel.java: Replace dummy search page with actual search URL.

KevinChuangCH committed:
DeveloperGuide.adoc: Remove extra blank line.

KevinChuangCH committed:
UserGuide.adoc: Add tags to the section I have written.

KevinChuangCH committed:
chuangchengheng.adoc: Update the contrivbution to User Guide and Developer Guide.

Nethergale committed:
Introduce stop-gap measure for inconsistent test case

Nethergale committed:
Remove unused imports

Nethergale committed:
Remove unused import

Nethergale committed:
Merge pull request #131 from Nethergale/feature/manageSocialMediaPlatform

Introduce stop-gap measure for inconsistent test case

Nethergale committed:
Merge pull request #130 from KevinChuangCH/Documentation

Nethergale committed:
Merge pull request #129 from KevinChuangCH/UpdateSearchFeature

KevinChuangCH committed:
Update documentation

KevinChuangCH committed:
DeveloperGuide.java: Correct mistake in title of the section.

Nethergale committed:
Add dummy profile page url

Nethergale committed:
Merge pull request #134 from Nethergale/master

Add dummy profile page url

Nethergale committed:
Add import of BrowserPanel

Nethergale committed:
Allow test to execute get loaded URL

Nethergale committed:
Allow test to get mappings of social media platform

Nethergale committed:
Remove expectedUrl assignment to actualUrl

Nethergale committed:
Replace dummy page with default page

Nethergale committed:
Remove unused import

Nethergale committed:
Remove unused variable in BrowserPanel

KevinChuangCH committed:
Merge pull request #133 from KevinChuangCH/Documentation

Documentation

KevinChuangCH committed:
ParserUtil.java: Add author for collate.

KevinChuangCH committed:
Person.java: Add author for collate.

KevinChuangCH committed:
Add author for collate.

KevinChuangCH committed:
UniqueTagList.java: Add author for collate.

KevinChuangCH committed:
chuangchengheng.adoc: Add link to collate files.

KevinChuangCH committed:
Merge pull request #137 from KevinChuangCH/Documentation

chuangchengheng.adoc: Add link to collate files.

KevinChuangCH committed:
Merge pull request #138 from KevinChuangCH/Collate

Collate

Nethergale committed:
Remove unused parameter for getExpectedUrl

Nethergale committed:
Remove unused variable in AddressBookSystemTest
Update parseUrl method in BrowserPanel

Nethergale committed:
Make minor change as commit

Nethergale committed:
Revert changes to parseUrl

Nethergale committed:
Merge pull request #136 from Nethergale/test/testingCI

Testing of CI checks

Nethergale committed:
Add important note to add platform command of user guide

Nethergale committed:
Add important note icon

Nethergale committed:
Update developer guide for add platform command
Rename SocialMediaPlatformBuilder to SocialMediaPlatformFactory

Nethergale committed:
Add more content to developer guide for add platform command

Nethergale committed:
Make minor corrections to developer guide and portfolio

Nethergale committed:
Merge pull request #139 from Nethergale/feature/docs/developerGuide

Update documentation

Nethergale committed:
Add test case for search command to address book parser test

Nethergale committed:
Add test case for add command alias

Nethergale committed:
Add test cases for search command parser

Nethergale committed:
Fix checkstyle errors

Nethergale committed:
Add various test cases for search command parser

Nethergale committed:
Add one more test for search command parser
Update portfolio and developer guide

Nethergale committed:
Merge pull request #140 from Nethergale/test/coverageImprovements

Improve code coverage

Nethergale committed:
Add collate files for group
Edit build.gradle to output jar with application name

Nethergale committed:
Merge pull request #141 from Nethergale/feature/docs/collate

Add collate files for group

Nethergale committed:
Change version to v1.5

Nethergale committed:
Update image file path for github io

Nethergale committed:
Update sequence diagram image with less spacing

Nethergale committed:
Add paragraph on data encryption

shadow2496 committed:
Fix merge conflicts

shadow2496 committed:
Merge branch 'master' into feature/model/login_facebook

shadow2496 committed:
Merge pull request #102 from shadow2496/feature/model/login_facebook

Model: made LoginCommand work limited to Facebook

shadow2496 committed:
Add annotations for collate

shadow2496 committed:
Add collated files

shadow2496 committed:
Merge pull request #143 from shadow2496/collate/shadow2496

Collate: added collated files

Nethergale committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B3/main into feature/releasePreparation

Nethergale committed:
Merge pull request #142 from Nethergale/feature/releasePreparation

Preparation of release v1.5
